packages/stores/src/derived-data/pools/metrics.ts

```diff
@@ -120,9 +123,38 @@ export class ObservablePoolsWithMetric {
         .get(this.chainId)
         .osmosis?.queryGammPools.getAllPools();
 
+      // Add all approved assets to a map for faster lookup.
+      const approvedAssets = new Map<string, boolean>();
+
+      /**
+       * Avoid unneeded calculation: skip adding approved assets if it's Frontier.
+       * Frontier will display all pools.
+       *  */
+      if (!this.isFrontier) {
+        [
+          ...this.assetStore.ibcBalances,
+          ...this.assetStore.nativeBalances,
+        ].forEach((asset) => {
+          approvedAssets.set(asset.balance.denom, true);
+        });
+      }
+
```

Comment:
IMO we should wrap the pools store with a filtered pools store that implements pool getter interface. Perhaps a "`AssetFilteredPoolsStore implements PoolsGetter`"

It feels "undry" to insert these distinctions in the pools store

Reply:
That's a good idea! I created a separate store for asset filtered pools. It will have to be inside derived data though, since the assets store depends on the queries store to fetch balances